### PR TITLE
Add optional support for empty MapSets

### PR DESCRIPTION
**Pull request by bernardd**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -359,6 +359,14 @@ Determines if fields in the changeset with `nil` values will be inserted as Dyna
 
 Determines if, during **Repo.update** or **Repo.update_all**, fields in the changeset with `nil` values will be removed from the record/s or set to the DynamoDB `null` value. This option is also available inline per query.
 
+**:empty_mapset_to_nil** :: boolean, *default:* `false`
+
+When a field contains a `DynamoDBSet` type and has a MapSet in it, by default the mapset must contain one or more values, otherwise an `insert` on the object will fail. This is due to DynamoDB not supporting empty sets. Setting this value to `true` will convert empty mapset values to `nil` before writting, allowing them to be written to DynamoDB as a null value.
```
Comment (alhambra1) [inline README.md:364]: Should "writting" be "writing"?

Comment (bernardd) [reply README.md:364]: Sure, if you want to spell things the "traditional" way.

File: lib/ecto_adapters_dynamodb.ex [hunk 2]
```diff
@@ -1517,45 +1514,39 @@ defmodule Ecto.Adapters.DynamoDB do
...
   # Decodes datetime, seemingly unhandled by ExAws Dynamo decoder
-  defp decode_type(val, type) do
-    if is_nil(val) do
-      val
-    else
-      case type do
-        t when t in [:utc_datetime_usec, :utc_datetime] ->
-          {:ok, dt, _offset} = DateTime.from_iso8601(val)
-          dt
+  defp decode_type(nil, DynamoDBSet, repo, opts), do: maybe_replace_nil_mapset(repo, opts)
```
Comment (darrenklein) [inline lib/ecto_adapters_dynamodb.ex:1531]: I like the way you broke the conditional/case into individual function definitions.

**Review by darrenklein — APPROVED** [review]
Looks good to me, I approve.

**Pull request merged**